src/lib/utilities/popup/types.ts

```diff
@@ -0,0 +1,44 @@
+// Popup Types
+// Note: these are a simple iteration based on the official docs.
+
+// Placement
+// https://floating-ui.com/docs/computePosition#placement
+type Placement =
+	| 'top'
+	| 'top-start'
+	| 'top-end'
+	| 'right'
+	| 'right-start'
+	| 'right-end'
+	| 'bottom'
+	| 'bottom-start'
+	| 'bottom-end'
+	| 'left'
+	| 'left-start'
+	| 'left-end';
+
+// Middleware
+interface Middleware {
+	/** Offset options: https://floating-ui.com/docs/offset */
+	offset?: number | Record<string, any>;
+	/** Shift options: https://floating-ui.com/docs/shift */
+	shift?: Record<string, any>;
+	/** Flip options: https://floating-ui.com/docs/flip */
+	flip?: Record<string, any>;
+	/** Arrow options: https://floating-ui.com/docs/arrow */
+	arrow?: Record<string, any>;
+}
```

Comment:
(VSCode is bugging out, hope that this works, it's meant to be the one where it imports from floating UI)

```suggestion
interface Middleware {
	/** Offset options: https://floating-ui.com/docs/offset */
	offset?: OffsetOptions;
	/** Shift options: https://floating-ui.com/docs/shift */
	shift?: ShiftOptions;
	/** Flip options: https://floating-ui.com/docs/flip */
	flip?: FlipOptions;
	/** Arrow options: https://floating-ui.com/docs/arrow */
	arrow?: ArrowOptions;
}
```